hazelcast/src/test/java/com/hazelcast/cache/eviction/ExtendedCacheExpirationTest.java

```diff
@@ -0,0 +1,154 @@
+import static com.hazelcast.test.OverridePropertyRule.set;
+import static java.lang.String.valueOf;
+import static java.util.Arrays.asList;
+import static org.junit.Assert.assertNull;
+
+@RunWith(Parameterized.class)
+@Parameterized.UseParametersRunnerFactory(HazelcastParallelParametersRunnerFactory.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class ExtendedCacheExpirationTest extends CacheTestSupport {
+
+    @Rule
+    public final OverridePropertyRule overrideTaskSecondsRule = set(PROP_TASK_PERIOD_SECONDS,
+            valueOf(Integer.MAX_VALUE));
+
+    @Parameterized.Parameters(name = "useSyncBackups:{0}")
+    public static Collection<Object[]> parameters() {
+        return asList(new Object[][]{
+                {true},
+                {false}
+        });
+    }
+
+    @Parameterized.Parameter(0)
+    public boolean useSyncBackups;
+
+    private static final int CLUSTER_SIZE = 2;
+    private final Duration THREE_SECONDS = new Duration(TimeUnit.SECONDS, 3);
+
+    protected TestHazelcastInstanceFactory factory;
+    protected HazelcastInstance[] instances = new HazelcastInstance[3];
+
+    @Override
+    protected HazelcastInstance getHazelcastInstance() {
+        return instances[0];
+    }
+
+    @Override
+    protected void onSetup() {
+        factory = createHazelcastInstanceFactory(CLUSTER_SIZE);
+        for (int i = 0; i < CLUSTER_SIZE; i++) {
+            instances[i] = factory.newHazelcastInstance(getConfig());
+        }
+    }
+
+    @Override
+    protected Config getConfig() {
+        return smallInstanceConfig();
+    }
+
+    @Override
+    protected void onTearDown() {
+        factory.shutdownAll();
+    }
+
+    @Test
+    public void test_backupOperationAppliesDefaultExpiryPolicy() {
+        HazelcastExpiryPolicy defaultExpiryPolicy = new HazelcastExpiryPolicy(THREE_SECONDS,
+                Duration.ZERO, Duration.ZERO);
+
+        CacheConfig cacheConfig = createCacheConfig(defaultExpiryPolicy);
+        final ICache cache = createCache(cacheConfig);
+
+        final int keyCount = 100;
+
+        for (int i = 0; i < keyCount; i++) {
+            cache.put(i, i);
+        }
+
+        // Check if all backup entries have applied the default expiry policy
+        for (int i = 1; i < CLUSTER_SIZE; i++) {
+            BackupAccessor backupAccessor = TestBackupUtils.newCacheAccessor(instances, cache.getName(), i);
+            for (int j = 0; j < keyCount; j++) {
+                TestBackupUtils.assertExpirationTimeExistsEventually(j, backupAccessor);
+            }
+        }
+
+        // terminate other nodes than number zero to cause backup promotion at the 0th member
+        for (int i = 1; i < CLUSTER_SIZE; i++) {
+            getNode(instances[i]).shutdown(true);
+        }
+
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() throws Exception {
```

Comment:
lambda style would be less verbose?

Reply:
3.12.z doesn't support lambda style.

Reply:
Ahh, sorry, didn't notice that this is 3.x.